MFVNeutralino/plugins/Vertexer.cc

```diff
@@ -34,2115 +34,2250 @@
-                h_miss_seed_track_npxlayers_minr->Fill(npxlayers, min_r);
-                tracks_missseed->push_back(ttk.track()); 
-            }
+  for (const reco::TrackRef& tk : *seed_track_refs) {
+    seed_tracks.push_back(tt_builder->build(tk));
+    seed_track_ref_map[tk] = seed_tracks.size() - 1;
+  }
 
-        }
+  const size_t ntk = seed_tracks.size();
+  if (verbose){
+    printf("n_seed_tracks: %5lu\n", ntk);
+    if (track_attachment)
+      printf("n_quality_tracks: %5lu\n", quality_track_refs->size());
+  }
+  //////////////////////////////////////////////////////////////////////
+  // Form seed vertices from all pairs of tracks whose vertex fit
+  // passes cuts.
+  //////////////////////////////////////////////////////////////////////
+
+  std::unique_ptr<reco::VertexCollection> vertices(new reco::VertexCollection);
+  std::unique_ptr<VertexerPairEffs> vpeffs(new VertexerPairEffs);
+  std::vector<std::pair<track_set, track_set>> vpeffs_tracks;
+
+  if (ntk == 0) {
+    if (verbose)
+      printf("no seed tracks -> putting empty vertex collection into event\n");
+    finish(event, seed_tracks, std::move(vertices), std::move(vpeffs), vpeffs_tracks);
+    return;
+  }
 
+  std::vector<size_t> itks(n_tracks_per_seed_vertex, 0);
+  auto try_seed_vertex = [&]() {
+    std::vector<reco::TransientTrack> ttks(n_tracks_per_seed_vertex);
+    for (int i = 0; i < n_tracks_per_seed_vertex; ++i)
+      ttks[i] = seed_tracks[itks[i]];
 
+    TransientVertex seed_vertex = kv_reco->vertex(ttks);
+    if (seed_vertex.isValid() && seed_vertex.normalisedChiSquared() < max_seed_vertex_chi2) {
+      vertices->push_back(reco::Vertex(seed_vertex));
 
+      if (verbose || histos) {
+        const reco::Vertex& v = vertices->back();
+        const double vchi2 = v.normalizedChi2();
+        const double vndof = v.ndof();
+        const double vx = v.position().x() - bsx;
+        const double vy = v.position().y() - bsy;
+        const double vz = v.position().z() - bsz;
+        const double phi = atan2(vy, vx);
+        const double rho = mag(vx, vy);
+        const double r = mag(vx, vy, vz);
+        if (verbose) {
+          printf("from tracks");
+          for (auto itk : itks)
+            printf(" %lu", itk);
+          printf(": vertex #%3lu: chi2/dof: %7.3f dof: %7.3f pos: <%7.3f, %7.3f, %7.3f>  rho: %7.3f  phi: %7.3f  r: %7.3f\n", vertices->size() - 1, vchi2, vndof, vx, vy, vz, rho, phi, r);
+        }
+        if (histos) {
+          for (auto it = v.tracks_begin(), ite = v.tracks_end(); it != ite; ++it)
+            h_seed_vertex_track_weights->Fill(v.trackWeight(*it));
+          h_seed_vertex_chi2->Fill(vchi2);
+          h_seed_vertex_ndof->Fill(vndof);
+          h_seed_vertex_x->Fill(vx);
+          h_seed_vertex_y->Fill(vy);
+          h_seed_vertex_rho->Fill(rho);
+          h_seed_vertex_phi->Fill(phi);
+          h_seed_vertex_z->Fill(vz);
+          h_seed_vertex_r->Fill(r);
+        }
+      }
     }
-
```

Comment:
Anyone know what this was? I don't recognize it, and it wasn't there in the pre-UL branches (and git blame isn't helping me right now). @pkotamnives was this for a study at some point, or something? Or @abbywarden do you know?